### DISCUSSION TITLE
Add `edgePadding` support to `animateToRegion()`

### DISCUSSION THREAD

**Discussion by go-sean-go** [react-native-maps/react-native-maps#4474] [Feature requests]
Currently, `fitToElements()` + `fitToSuppliedMarkers()` + `fitToCoordinates()` all support adjustments to `edgePadding`, allowing the map to appear 'centered' despite overlapping UI elements (or whatever other use case).

(My use case right now is that I have a large 'bottom sheet' element on the map.)

The primary workflow (for me) that is unsupported by the current `fitTo...` methods is using `animateToRegion()` combined with the user's GPS dot. (I'm not sure how to use the insets of my UI elements to adjust the `Region` object passed to `animateToRegion()`.)

**Comment by go-sean-go** [comment]
For future readers, until this is supported (or if it is never supported), after about ~30 mins of thinking & tinkering, I was able to put something together that I think is fairly durable ASSUMING your `mapInsets` are accurate:

```
if (userLocation) {
  const screenHeight = Dimensions.get('screen').height; // my overlapping UI elements are all on the Y-axis; map width is edge-to-edge
  const bottomInsets = mapInsets.bottom; // again, assumes you have accurate mapInsets
  const usableBottomInsets =
    typeof bottomInsets === 'number' ? bottomInsets : 0;
  const bottomRatio = usableBottomInsets / screenHeight; // typescript stuff

  mapRef.current?.animateToRegion({
    ...userLocation, // GPS coords + lat/long deltas that look visually appearing
    latitude:
      userLocation.latitude -
      bottomRatio * userLocation.latitudeDelta * 0.5, // the magic - use the `latitudeDelta` as a proxy for the screen height as a unit of latitude, then multiple that unit by the amount of the screen your UI elements cover (`bottomRatio`) and keep the adjusted 'dot' in the center by using 1/2 of that value
  });
}
```

**Reply by rawatnaresh** [reply]
I think that is the only way. They have done something similar in the [examples](https://github.com/react-native-maps/react-native-maps/blob/master/example/src/examples/AnimatedViews.tsx#L263-L296)
https://github.com/react-native-maps/react-native-maps/blob/master/example/src/examples/AnimatedViews.tsx#L263-L296

**Reply by go-sean-go** [reply]
Interesting...then assuming the same technical solution could be used for support for `animateToRegion()` (under the hood/within the library), it might not provide a 'better' solution than what I discovered per se - but it _would_ save future devs from the hassle of repeating the same investigation/work over and over (and probably save a few from a bug or two).

**Reply by rawatnaresh** [reply]
yeah also if you would like to animate your map region based on the changing height of your bottom sheet, you can try this.

```js
import MapView, {Marker, PROVIDER_GOOGLE} from 'react-native-maps';
import Animated, {
  Extrapolate,
  interpolate,
  useAnimatedProps,
  useSharedValue,
} from 'react-native-reanimated';

const AnimatedMap = Animated.createAnimatedComponent(MapView);

...

  const animatedIndex = useSharedValue(0);
  const animatedProps = useAnimatedProps(() => ({
    region: {
      latitude: interpolate(
        animatedIndex.value,
        [0,  SOME_HEIGHT],
        coor.latitude, coor.latitude - LATITUDE_DELTA * 0.4],
        Extrapolate.CLAMP,
      ),
      longitude: coor.longitude,
      latitudeDelta: LATITUDE_DELTA,
      longitudeDelta: LONGITUDE_DELTA,
    },
  }));
  
  ...
  
  <AnimatedMap
        ref={mapRef}
        zoomEnabled={true}
        provider={PROVIDER_GOOGLE}
        animatedProps={animatedProps}
        style={styles.mapContainer}>
  </AnimatedMap>

```

**Reply by calebpanza** [reply]
Does this result in animating the bottom inset, though? I am trying to get the bottom inset to update as my sheet updates, though it's a bit "rigid" and jumps around quite a bit
